rpmio/macro.c

```diff
@@ -864,6 +865,27 @@ doFoo(MacroBuf mb, int negate, const char * f, size_t fn,
 	if ((b = strrchr(buf, '/')) != NULL)
 	    *b = '\0';
 	b = buf;
+    } else if (STREQ("shrink", f, fn)) {
+	/*
+	 * shrink body by removing all leading and trailing whitespaces and
+	 * reducing intermediate whitespaces to a single space character.
+	 */
+	int i, j, k, was_space = 0;
+	for (i = 0, j = 0, k = (int)strlen(buf); i < k; ) {
+	    if (isspace((int)(buf[i]))) {
+		was_space = 1;
+		i++;
+		continue;
+	    } else if (was_space) {
+		was_space = 0;
+		if (j > 0) /* remove leading blanks at all */
+		    buf[j++] = ' ';
+		/* fallthrough */
```

Comment:
The /* fallthrough */ comment seems out of place, there's no switch-case here. A leftover porting artifact maybe?